src/server/scripts/EasternKingdoms/zone_redridge_mountains.cpp

```diff
@@ -113,147 +138,612 @@ class npc_big_earl : public CreatureScript
     SAY_ALEX_PUSH          = 4,
     SAY_WORKERS_HO         = 0,
-    SAY_MATTHEW_IM_PUSHING = 1
+    SAY_MATTHEW_IM_PUSHING = 1,
+
+    POINT_NEAR_BOULDER  = 1,
+    POINT_NEAR_WATER    = 2,
+    POINT_UP_PATH       = 3,
+};
+
+constexpr Position EttinPathToWater[] =
+{
+    { -9270.707f, -2299.5933f, 69.642654f },
+    { -9271.457f, -2309.5933f, 69.892654f },
+    { -9279.707f, -2319.0933f, 66.892654f },
+    { -9303.957f, -2330.5933f, 61.642654f },
+    { -9324.86f,  -2338.94f,   61.2445f  }
 };
 
-class npc_bridge_worker_alex : public CreatureScript
+constexpr Position EttinPathUpHill[] =
+{
+    { -9308.439f, -2330.975f, 61.716003f },
+    { -9292.939f, -2325.475f, 62.716003f },
+    { -9276.939f, -2316.725f, 68.966f },
+    { -9260.689f, -2312.725f, 75.216f },
+    { -9237.689f, -2313.225f, 78.716f },
+    { -9214.52f,  -2334.01f,  83.6875f }
+};
+
+class npc_redridge_huge_boulder : public CreatureScript
 {
 public:
-    npc_bridge_worker_alex() : CreatureScript("npc_bridge_worker_alex") { }
+    npc_redridge_huge_boulder() : CreatureScript("npc_redridge_huge_boulder") { }
 
-    struct npc_bridge_worker_alexAI : public ScriptedAI
+    struct npc_redridge_huge_boulderAI : public CreatureAI
     {
-        npc_bridge_worker_alexAI(Creature* creature) : ScriptedAI(creature) { }
+        npc_redridge_huge_boulderAI(Creature* creature) : CreatureAI(creature) { }
 
         void Reset() override
         {
             _events.Reset();
             _oslowGUID.Clear();
+            _alexGUID.Clear();
             _trentGUID.Clear();
-            _dimitriGUID.Clear();
+            _dmitriGUID.Clear();
             _jessGUID.Clear();
             _danielGUID.Clear();
             _matthewGUID.Clear();
-            _events.ScheduleEvent(EVENT_STORE_GUIDS, Seconds(5));
-            _events.ScheduleEvent(EVENT_OSLOW_SAY, Seconds(40), Seconds(50));
-            _events.ScheduleEvent(EVENT_ALEX_SAY, Seconds(22), Seconds(30));
+            _events.ScheduleEvent(EVENT_STORE_GUIDS, 5s);
+        }
+
+        void SpellHit(WorldObject* /*caster*/, SpellInfo const* spell) override
+        {
+            if (spell->Id != SPELL_LIFT_BOULDER_RIDE)
+                return;
+
+            _events.Reset();
+            _events.ScheduleEvent(EVENT_REPOSITION, 500ms);
         }
 
         void UpdateAI(uint32 diff) override
         {
             _events.Update(diff);
 
-            if (!UpdateVictim())
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                while (uint32 eventId = _events.ExecuteEvent())
+                switch (eventId)
                 {
-                    switch (eventId)
-                    {
                     case EVENT_STORE_GUIDS:
-                        if (Creature* oslow = me->FindNearestCreature(NPC_OSLOW, 5.0f, true))
+                        if (Creature const* oslow = me->FindNearestCreatureWithOptions(10.0f,  { .StringId = "npc_redridge_foreman_oslow_under_rock"sv }))
```

Comment:
is the stringid here required? In general we only use stringids if we cannot get the creature via entry only (e.g. bcuz the same entry is spawned 10 times and we want to use a specific one)